paper/paper.md

```diff
@@ -94,11 +107,27 @@ override fun simulate(node: NodeType): Double {
 }
 ```
 
+![Reversi Heuristic. \label{fig:reversi-heu} ](reversi-heu.png?raw=true "Title")
+
 # Results
 
 When the MCTS solver is accurately selecting the optimal solutions, it will continue to compel the agent to explore in the optimal subset of the state space, and reduce its exploration in the non-optimal subset of the state space. We provide a quick example in the MDP Domain of GridWorld, detailed in [@Liu:2021]. The cumulative number of visits corresponding to the optimal policy is proportionally increasing with respect to the number of MCTS iterations. Whereas for non-optimal solutions, the cumulative visits are significantly lower because the solver will avoid visiting the non-optimal state subspace. 
 
-![Convergence of visits](gw_visits.png?raw=true "Title")
+![Convergence of visits.](gw_visits.png?raw=true "Title")
+
+# Real World Applications
+
+*mctreesearch4j* is designed to run on the JVM, and therefore is useful in a research setting where JVM is intended to be used alongside either the implementation of the game or the core AI engine. Typically, for state-of-the-art AI, MCTS is used in combination with other Deep Learning algorithms, where MCTS drives the exploration of hypothetical games to be played out in order for the agent to make better decisions.
+
+MCTS is typically combined with Deep Learning strategies to achieve state-of-the-art performance for complex games whose solutions are typically intractable via standard tree-search methods. For example in AlphaGo [silver:2016], MCTS is combined with two types of Neural Networks, namely Value Networks and Policy Networks, to achieve state-of-the-art performance in the game of Go, never achieved before. Similarly, in the area of control theory, MCTS acting in combination with Value and Policy Networks to produce state and action space distributions, have shown impressive performance in probabilistic planning for self-driving cars, as demonstrated in Tesla’s AI day event [tesla:2021]. The utility of MCTS illustrates the acceptance of MCTS into the standard repetoire of both probablisitc and deterministic planning. And as the need for MCTS grows among industry and research, the impetus to create MCTS software libraries that are standardized, reliable, and can integrate well into general software ecosystems, such as JVM, is increasing.
+
+One key industry application of *mctreesearch4j* is the fact that it is a JVM software, which allows it to integrate into both legacy and modern JVM software. One key application can be in mobile gaming, which has a large developer base in Android, largely written in Kotlin within the JVM. *mctreesearch4j*’s lightweight design with guaranteed reliability and JVM compatibility, tested extensively on both Java 8 and Java 11, make it a prime candidate for robust AI in turn-based games on mobile devices.
```

Comment:
Consider removing this paragraph. It arguably conveys too much of a "sales" attitude. Also, consider carefully re-reading the paper and edit for a neutral writing style.

Reply:
> I added some comments. Also, the setup is a bit cumbersome. To build the project, one apparently needs signing and publishing credentials, and when build the examples, one needs to move the lib to avoid this error message:
> 
> ```
> Execution failed for task ':compileKotlin'.
> > Could not resolve all files for configuration ':compileClasspath'.
>    > Could not find ca.aqtech.app:app-1.0-SNAPSHOT:.
>      Searched in the following locations:
>        - file:/.../mctreesearch4j/app/libs/app-1.0-SNAPSHOT.jar
> ```

The signing and publishing credentials are tied to a GPG key associated with this repository. Maven central requires all artifacts be signed with a GPG key, this is automated in Github .workflows. You may run a compile without signing the .jar via
```
gradle clean build -x uploadArchives
```
This will skip the signing step, and allow for a local unsigned .jar to be compiled for testing. We added this to the README.md in https://github.com/aqtech-ca/mctreesearch4j/pull/71

Reply:
@TimKam We actually added this section in because @slim-patchy requested a discussion about potential applications in commercial setting. We think it is somewhat valuable to the current paper, however, we are also perfectly fine with removing the section as well. I would ask @slim-patchy to weigh in on this to give further clarification.